overlord/ifacestate/apparmorprompting/common/common.go

```diff
@@ -336,18 +336,337 @@ func abstractPermissionsToAppArmorFilePermissions(iface string, permissions []st
 	return filePerms, nil
 }
 
-var allowablePathPatternRegexp = regexp.MustCompile(`^(/|(/[^/*{}]+)*(/\*|(/\*\*)?(/\*\.[^/*{}]+)?)?)$`)
+var (
+	// The following must be escaped as literals in a path pattern:
+	problematicChars = `\[\]`
+	// A single safe or escaped unsafe char in a path
+	safePathChar = fmt.Sprintf(`([^%s]|\\[%s])`, problematicChars, problematicChars)
+
+	// The following matches valid path patterns
+	allowablePathPattern       = fmt.Sprintf(`^/%s*$`, safePathChar)
+	allowablePathPatternRegexp = regexp.MustCompile(allowablePathPattern)
```

Comment:
Good point, I agree that inlining makes the most sense here, given it's just two characters left (albeit escaped).

Reply:
Actually, it could all be inlined into:

```go
var allowablePathPatternRegexp = regexp.MustCompile(`^/([^\[\]]|\\[\[\]])*$`)
```

I think this is nicer than having a bunch of variables and string formatting.